### PR TITLE
Pr/durations measurement

### PR DESCRIPTION
**Pull request by ankuns**
Implemented measurement of:
- total time spent while listening
- total tome spent while really receiving
- total time spent while transmitting.

Implementation is based on timer_coord. Use of additional TIMER is considered unwanted thus software read of hp_timer is performed to timestamp some operations.

### CODE REVIEW

**Comment by hubertmis** [comment]
:+1:

File: src/nrf_802154_core.c [hunk 1]
```diff
@@ -138,6 +139,21 @@ static nrf_802154_timer_t m_rx_prestarted_timer;
...
 static nrf_802154_coex_tx_request_mode_t m_coex_tx_request_mode;
+
+#if NRF_802154_TOTAL_TIMES_MEASUREMENT_ENABLED
+#if !NRF_802154_FRAME_TIMESTAMP_ENABLED
+#error NRF_802154_FRAME_TIMESTAMP_ENABLED == 0 when NRF_802154_TOTAL_TIMES_MEASUREMENT_ENABLED != 0
+#endif
+#if NRF_802154_DISABLE_BCC_MATCHING
+#error NRF_802154_DISABLE_BCC_MATCHING != 0 when NRF_802154_TOTAL_TIMES_MEASUREMENT_ENABLED != 0
+#endif
+#endif
```
Comment (jciupis) [inline src/nrf_802154_core.c]: FCS: `#endif // NRF_802154_TOTAL_TIMES_MEASUREMENT_ENABLED`

File: src/nrf_802154_core.c [hunk 2]
```diff
@@ -138,6 +139,21 @@ static nrf_802154_timer_t m_rx_prestarted_timer;
...
+
+#if NRF_802154_TOTAL_TIMES_MEASUREMENT_ENABLED
+static uint32_t m_listening_start_hp_timestamp;
+
```
Comment (jciupis) [inline src/nrf_802154_core.c]: This newline can be removed

File: src/nrf_802154_core.c [hunk 3]
```diff
@@ -689,6 +705,58 @@ static void operation_terminated_notify(radio_state_t state, bool receiving_psdu
...
+    }
+}
+
+#endif
+
+static void trx_abort(void)
+{
+#if (NRF_802154_TOTAL_TIMES_MEASUREMENT_ENABLED)
+    trx_state_t trx_state = nrf_802154_trx_state_get();
+
```
Comment (jciupis) [inline src/nrf_802154_core.c]: Please go through these `#endif`s and remove obsolete newlines wherever needed

File: src/nrf_802154_core.c [hunk 4]
```diff
@@ -1470,19 +1549,106 @@ void nrf_802154_trx_go_idle_finished(void)
...
+            uint32_t t_listening = listening_end_timestamp - m_listening_start_hp_timestamp;
+
+            m_listening_start_hp_timestamp = listening_end_timestamp;
+
+            nrf_802154_stat_totals_increment(total_listening_time, t_listening);
+        }
+        else
+        {
+            /* To little time passed since m_listening_start_hp_timestamp, we don't know
```
Comment (czeslawmakarski) [inline src/nrf_802154_core.c]: Typo: 'Too'

File: src/nrf_802154_core.c (hunk 2 above)
Comment (ankuns) [reply src/nrf_802154_core.c]: You think so, I think so also, pretty.sh doesn't think as us. pretty.sh doesn't think at all.

File: src/nrf_802154_core.c [hunk 5]
```diff
@@ -869,10 +937,21 @@ static void rx_init(bool disabled_was_triggered)
...
 
+#if NRF_802154_TOTAL_TIMES_MEASUREMENT_ENABLED
+    m_listening_start_hp_timestamp = nrf_802154_hp_timer_current_time_get();
+#endif
+
 #if (NRF_802154_FRAME_TIMESTAMP_ENABLED)
+#if (NRF_802154_TOTAL_TIMES_MEASUREMENT_ENABLED)
+    // Configure the timer coordinator to get a timestamp of the END event which
+    // fires several cycles after CRCOK or CRCERROR events.
+    nrf_802154_timer_coord_timestamp_prepare(
```
[the remark this answers is not in the document]
Comment (ankuns) [reply src/nrf_802154_core.c]: Discussed F2F

File: src/nrf_802154_core.c (hunk 2 above)
Comment (jciupis) [reply src/nrf_802154_core.c]: Valid point.

File: src/nrf_802154_stats.c [hunk 6]
```diff
@@ -34,10 +34,14 @@
 #include "nrf_802154_stats.h"
 
 #define NUMBER_OF_STAT_COUNTERS (sizeof(nrf_802154_stat_counters_t) / sizeof(uint32_t))
+#define NUMBER_OF_STAT_TOTALS   (sizeof(nrf_802154_stat_totals_t) / sizeof(uint64_t))
```
[the remark this answers is not in the document]
Comment (ankuns) [reply src/nrf_802154_stats.c]: Please note that `nrf_802154_stat_totals_t` is a `struct`, so it cannot be indexed as an array.

File: src/nrf_802154_core.c (hunk 3 above)
Comment (ankuns) [reply src/nrf_802154_core.c]: Uncrustify requires them

**Pull request merged**